onnxruntime/core/providers/xnnpack/nn/conv_base.cc

```diff
@@ -81,6 +81,28 @@ Status CreateXnnpackKernel(const ConvAttributes& conv_attrs,
         foutput_min, foutput_max, flags,
         code_cache, weights_cache,
         &p);
+  } else if (conv_type == OpComputeType::op_compute_type_fp16) {
+    const auto* B_data = Bias ? Bias->Data<MLFloat16>() : nullptr;
+    // 65504 is the max value of float16
+    // https://en.wikipedia.org/wiki/Half-precision_floating-point_format
+    float output_min = clip_min_max ? onnxruntime::math::floatToHalf(clip_min_max->first) : -65504.0;
```

Comment:
why do we assign a half value to float?
BTW, we can easy to use a MLFloat(value) to construct a half value.

Reply:
Add the suffix 'f' after a float constant to avoid some compiler warnings.

Reply:
> why do we assign a half value to float? BTW, we can easy to use a MLFloat(value) to construct a half value.

xnn_create_deconvolution2d_nhwc_f16's out_min type is a float

Reply:
Got it.   onnxruntime::math::floatToHalf is somehow misleading though